tests/app/data/test_schema_validation.py

```diff
@@ -28,27 +29,89 @@ def test_schema(self):
         for file in files:
 
             json_file = open(os.path.join(settings.EQ_SCHEMA_DIRECTORY, file), encoding="utf8")
+            json_to_validate = json.load(json_file)
+
+            errors.extend(self.validate_json_against_schema(file, json_to_validate, schema))
 
-            error = self.validate_json_against_schema(file, json_file, schema)
+            errors.extend(self.validate_schema_contains_valid_routing_rules(file, json_to_validate))
 
-            if error:
-                errors.append(error)
+            errors.extend(self.validate_schema_containing_routing_rules_has_default_if_not_all_answers_routed(file, json_to_validate))
 
         if errors:
             for error in errors:
                 logger.error(error)
 
             self.fail("{} Schema Validation Errors.".format(len(errors)))
 
+
     @staticmethod
-    def validate_json_against_schema(file, json_file, schema):
+    def validate_json_against_schema(file, json_to_validate, schema):
         try:
-            json_to_validate = json.load(json_file)
             validate(json_to_validate, schema)
+            return []
         except jsonschema.exceptions.ValidationError as e:
-            return "Schema Validation Error! File [{}] does not validate against schema. Error [{}]".format(file, e)
+            return ["Schema Validation Error! File [{}] does not validate against schema. Error [{}]".format(file, e)]
         except JSONDecodeError as e:
-            return "JSON Parse Error! Could not parse [{}]. Error [{}]".format(file, e)
+            return ["JSON Parse Error! Could not parse [{}]. Error [{}]".format(file, e)]
+
+    @classmethod
+    def validate_schema_contains_valid_routing_rules(cls, file, json_to_validate):
+
+        errors = []
+
+        for block in SchemaHelper.get_blocks(json_to_validate):
+            if 'routing_rules' in block and len(block['routing_rules']) > 0:
+                for rule in block['routing_rules']:
+                    is_goto_rule = 'goto' in rule and 'id' in rule['goto'].keys()
+                    if is_goto_rule:
+                        if rule['goto']['id'] == 'summary':
+                            continue
+
+                        if not cls.has_block(json_to_validate, rule['goto']['id']):
+                            errors.append("Schema Error. File[{}] Routing rule routes to invalid block [{}]".format(file, rule['goto']['id']))
+
+        return errors
+
+    @staticmethod
+    def validate_schema_containing_routing_rules_has_default_if_not_all_answers_routed(file, json_to_validate):
```

Comment:
I think `validate_routing_rules_has_default_if_not_all_answers_routed` says the same in a few less words